src/util/assert.rs

```diff
@@ -0,0 +1,15 @@
+/// macro for compile-time const assertions
+macro_rules! const_assert {
+    ($($list:ident : $ty:ty),* => $expr:expr) => {{
+        struct Assert<$(const $list: $ty,)*>;
+        impl<$(const $list: $ty,)*> Assert<$($list,)*> {
+            const OK: u8 = 0 - !($expr) as u8;
```

Comment:
Addressed in [025c331](https://github.com/gendx/lzma-rs/pull/79/commits/025c33142e265080bafc951e256f7a0fb5ede3b6) but this will bump MSRV from 1.51 to 1.57.

Reply:
Bumping to 1.57 should be fine by now. I've also cut a new version 0.3.0 (https://github.com/gendx/lzma-rs/commit/da82bd1182993a07db035e285a8f8a87d32a30c2), so the 0.2.x branch is still available for compilers < 1.57.